runtime/compiler/optimizer/TreeLowering.cpp

```diff
@@ -1120,8 +1104,13 @@ LoadArrayElementTransformer::lower(TR::Node* const node, TR::TreeTop* const tt)
    ///////////////////////////////////////
    // 9. Create ificmpne node that checks classFlags
 
+   TR::SymbolReference* const vftSymRef = comp->getSymRefTab()->findOrCreateVftSymbolRef();
+
+   TR::Node *vftNode = TR::Node::createWithSymRef(node, TR::aloadi, 1, anchoredArrayBaseAddressNode, vftSymRef);
+   TR::Node *testIsArrayClassNullRestrictedNode = comp->fej9()->testIsArrayClassNullRestrictedType(vftNode);
+
    // The branch destination will be set up later
-   TR::Node *ifNode = comp->fej9()->checkArrayCompClassPrimitiveValueType(anchoredArrayBaseAddressNode, TR::ificmpne);
+   TR::Node *ifNode = TR::Node::createif(TR::ificmpne, testIsArrayClassNullRestrictedNode, TR::Node::iconst(0));
```

Comment:
Could the comments before this method be updated to reflect this change that it tests `J9ClassArrayIsNullRestricted` flag now?

```
 |ificmpne ----->---------------------------+-----------+
 |   iand                                   |           |
 |      iloadi  <isClassFlags>              |           |
 |      ...                                 |           |
 |      iconst J9ClassIsPrimitiveValueType  |           |
 |   iconst 0  
```

Reply:
Thanks for catching that!

Reply:
Fixed in commit 686080c6d07076a61ed5078a017afff78d15e234